go/pkg/client/bytestream.go

```diff
@@ -26,33 +48,56 @@ func (c *Client) WriteBytes(ctx context.Context, name string, data []byte) error
 	return err
 }
 
+// WriteBytesWithOffset uploads a byte slice with a given resource name to the CAS
+// at an arbitrary offset.
+func (c *Client) WriteBytesWithOffset(ctx context.Context, name string, data []byte, opts ...ByteStreamWriteOption) (int64, error) {
+	ue := uploadinfo.EntryFromBlob(data)
+	ch, err := chunker.New(ue, false, int(c.ChunkMaxSize))
+	if err != nil {
+		return 0, err
+	}
+	return c.writeChunked(ctx, name, ch, opts...)
+}
+
 // writeChunked uploads chunked data with a given resource name to the CAS.
-func (c *Client) writeChunked(ctx context.Context, name string, ch *chunker.Chunker) (int64, error) {
+func (c *Client) writeChunked(ctx context.Context, name string, ch *chunker.Chunker, rawOpts ...ByteStreamWriteOption) (int64, error) {
+	if len(rawOpts) == 1 {
+		return 0, errors.Errorf("Missing an arugment")
+	}
+	opts := &ByteStreamWriteOpts{}
+	var resume bool
+	for _, o := range rawOpts {
+		resume = true
+		o.Apply(opts)
+	}
+
 	var totalBytes int64
 	closure := func() error {
 		// Retry by starting the stream from the beginning.
 		if err := ch.Reset(); err != nil {
 			return errors.Wrap(err, "failed to Reset")
 		}
 		totalBytes = int64(0)
-		// TODO(olaola): implement resumable uploads.
```

Comment:
Hmm, this implementation is not fully resumable since retries will resend all the chunks. A true resumable implementation would need to track the offset across retries to avoid resending successfully sent chunks. If that is not in the scope of your change, you could simply clarify that in the doc comment (that the new option allows to upload to an arbitrary offset,  but retries still resend from the specified offset).

Reply:
Done.